### PR TITLE
[improvement](executor)Add tag property for workload group

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes

Add tag property for workload group, the workload groups within same tag 's hard limit sum value can not exceeds 100%.
User can group ```workload group``` by Resource tag.

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/resource/workloadgroup/WorkloadGroup.java [hunk 1]
```diff
@@ -71,14 +71,18 @@ public class WorkloadGroup implements Writable, GsonPostProcessable {
 
     public static final String SPILL_THRESHOLD_HIGH_WATERMARK = "spill_threshold_high_watermark";
 
+    public static final String TAG = "tag";
+    public static final String TAG_DEFAULT_VALUE = "default_tag";
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/resource/workloadgroup/WorkloadGroup.java:75]: rename to empty_tag

Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/resource/workloadgroup/WorkloadGroup.java:75]: Do not use this logic to manage workload group without tag. 
If a workload's tag is empty, then it is on all be.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateWorkloadGroupStmt.java [hunk 2]
```diff
@@ -69,6 +72,15 @@ public void analyze(Analyzer analyzer) throws UserException {
         if (properties == null || properties.isEmpty()) {
             throw new AnalysisException("Resource group properties can't be null");
         }
+
+        String wgTag = properties.get(WorkloadGroup.TAG);
+        if (!StringUtils.isEmpty(wgTag)) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateWorkloadGroupStmt.java:77]: the tag value should be trimed.
And do not use default_tag for workload group without tag.

File: fe/fe-core/src/main/java/org/apache/doris/resource/workloadgroup/WorkloadGroup.java [hunk 3]
```diff
@@ -142,6 +146,9 @@ private WorkloadGroup(long id, String name, Map<String, String> properties, long
             }
             this.properties.put(SPILL_THRESHOLD_HIGH_WATERMARK, highWatermarkStr);
         }
+        if (properties.containsKey(TAG)) {
+            this.properties.put(TAG, properties.get(TAG).toLowerCase());
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/resource/workloadgroup/WorkloadGroup.java:150]: trim the tag value

File: fe/fe-core/src/main/java/org/apache/doris/resource/workloadgroup/WorkloadGroup.java [hunk 4]
```diff
@@ -431,6 +454,13 @@ public void getProcNodeData(BaseProcResult result, QueryQueue qq) {
                 row.add(qq == null ? "0" : String.valueOf(qq.getCurrentRunningQueryNum()));
             } else if (QueryQueue.WAITING_QUERY_NUM.equals(key)) {
                 row.add(qq == null ? "0" : String.valueOf(qq.getCurrentWaitingQueryNum()));
+            } else if (TAG.equals(key)) {
+                String val = properties.get(key);
+                if (StringUtils.isEmpty(val)) {
+                    row.add(TAG_DEFAULT_VALUE);
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/resource/workloadgroup/WorkloadGroup.java:460]: just leave it empty

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/MetadataGenerator.java [hunk 5]
```diff
@@ -431,10 +431,11 @@ private static TFetchSchemaTableDataResult workloadGroupsMetadataResult(TSchemaT
             trow.addToColumnValue(new TCell().setLongVal(Long.valueOf(rGroupsInfo.get(10))));
             // min remote scan thread num
             trow.addToColumnValue(new TCell().setLongVal(Long.valueOf(rGroupsInfo.get(11))));
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/MetadataGenerator.java:433]: We miss some logic?
When publish topic to BE, we should use tag to match the Backends?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateWorkloadGroupStmt.java (hunk 2 above)
Comment (wangbo) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CreateWorkloadGroupStmt.java:77]: tag name is checked in CreateWorkloadGroupStmt using FeNameFormat.checkCommonName

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/MetadataGenerator.java (hunk 5 above)
Comment (wangbo) [reply fe/fe-core/src/main/java/org/apache/doris/tablefunction/MetadataGenerator.java:433]: We can consider it when support multi process.

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by mrhhsg — APPROVED** [review]
LGTM

**Pull request merged**